openpype/hosts/maya/api/lib.py

```diff
@@ -172,6 +174,223 @@ def maintained_selection():
             cmds.select(clear=True)
 
 
+def reload_all_udim_tile_previews():
+    """Regenerate all UDIM tile preview in texture file"""
+    texture_files = cmds.ls(type="file")
+    if texture_files:
+        for texture_file in texture_files:
+            if cmds.getAttr("{}.uvTilingMode".format(texture_file)) > 0:
+                cmds.ogs(regenerateUVTilePreview=texture_file)
+    cmds.ogs(reloadTextures=True)
+
+
+@contextlib.contextmanager
+def panel_camera(panel, camera):
+    """Set modelPanel's camera during the context.
+
+    Arguments:
+        panel (str): modelPanel name.
+        camera (str): camera name.
+
+    """
+    original_camera = cmds.modelPanel(panel, query=True, camera=True)
+    try:
+        cmds.modelPanel(panel, edit=True, camera=camera)
+        yield
+    finally:
+        cmds.modelPanel(panel, edit=True, camera=original_camera)
+
+
+def render_capture_preset(preset):
+    """Capture playblast with a preset.
+
+    To generate the preset use `generate_capture_preset`.
+
+    Args:
+        preset (dict): preset options
+
+    Returns:
+        str: Output path of `capture.capture`
+    """
+
+    # Force a refresh at the start of the timeline
+    # TODO (Question): Why do we need to do this? What bug does it solve?
+    #   Is this for simulations?
+    cmds.refresh(force=True)
+    refresh_frame_int = int(cmds.playbackOptions(query=True, minTime=True))
+    cmds.currentTime(refresh_frame_int - 1, edit=True)
+    cmds.currentTime(refresh_frame_int, edit=True)
+
+    if os.environ.get("OPENPYPE_DEBUG") == "1":
+        log.debug(
+            "Using preset: {}".format(
+                json.dumps(preset, indent=4, sort_keys=True)
+            )
+        )
+
+    # not supported by `capture` so we pop it off of the preset
+    reload_textures = preset["viewport_options"].pop("reloadTextures", True)
```

Comment:
Should we make a copy of the input `preset` dict so that we're not changing the input argument `dict` which might be confusing for someone calling this function that their `preset` dict passed to the function has suddenly changed.

```python
preset = copy.deepcopy(preset)
```

That would solve it.